ecosystem/typescript/sdk/src/transaction_builder/builder_utils.ts

```diff
@@ -0,0 +1,179 @@
+    case "bool":
+      return new TypeTagBool();
+    case "u8":
+      return new TypeTagU8();
+    case "u64":
+      return new TypeTagU64();
+    case "u128":
+      return new TypeTagU128();
+    case "address":
+      return new TypeTagAddress();
+    default:
+      throw new Error("Unknown type tag.");
+  }
+}
+
+export function serializeArg(argVal: any, argType: TypeTag, serializer: Serializer) {
+  if (argType instanceof TypeTagBool) {
+    assertType(argVal, "boolean");
+    serializer.serializeBool(argVal);
+    return;
+  }
+  if (argType instanceof TypeTagU8) {
+    assertType(argVal, "number");
+    serializer.serializeU8(argVal);
+    return;
+  }
+  if (argType instanceof TypeTagU64) {
+    assertType(argVal, ["number", "bigint"]);
+    serializer.serializeU64(argVal);
+    return;
+  }
+  if (argType instanceof TypeTagU128) {
+    assertType(argVal, ["number", "bigint"]);
+    serializer.serializeU128(argVal);
+    return;
+  }
+  if (argType instanceof TypeTagAddress) {
+    let addr: AccountAddress;
+    if (typeof argVal === "string") {
+      addr = AccountAddress.fromHex(argVal);
+    } else if (argVal instanceof AccountAddress) {
+      addr = argVal;
+    } else {
+      throw new Error("Invalid account address.");
+    }
+    addr.serialize(serializer);
+    return;
+  }
+  if (argType instanceof TypeTagVector) {
+    if (!(argVal instanceof Array)) {
+      throw new Error("Invalid vector args.");
+    }
+
+    serializer.serializeU32AsUleb128(argVal.length);
+
+    argVal.forEach((arg) => serializeArg(arg, argType.value, serializer));
+    return;
+  }
+  throw new Error("Unsupported arg type.");
+}
+
+export function argToTransactionArgument(argVal: any, argType: TypeTag): TransactionArgument {
+  if (argType instanceof TypeTagBool) {
+    assertType(argVal, "boolean");
+    return new TransactionArgumentBool(argVal);
+  }
+  if (argType instanceof TypeTagU8) {
+    assertType(argVal, "number");
+    return new TransactionArgumentU8(argVal);
+  }
+  if (argType instanceof TypeTagU64) {
+    assertType(argVal, ["number", "bigint"]);
+    return new TransactionArgumentU64(argVal);
+  }
+  if (argType instanceof TypeTagU128) {
+    assertType(argVal, ["number", "bigint"]);
+    return new TransactionArgumentU128(argVal);
+  }
+  if (argType instanceof TypeTagAddress) {
+    let addr: AccountAddress;
+    if (typeof argVal === "string") {
+      addr = AccountAddress.fromHex(argVal);
+    } else if (argVal instanceof AccountAddress) {
+      addr = argVal;
+    } else {
+      throw new Error("Invalid account address.");
+    }
+    return new TransactionArgumentAddress(addr);
+  }
+  if (argType instanceof TypeTagVector && argType.value instanceof TypeTagU8) {
+    if (!(argVal instanceof Uint8Array)) {
+      throw new Error(`${argVal} should be an instance of Uint8Array`);
+    }
+    return new TransactionArgumentU8Vector(argVal);
+  }
+
+  throw new Error("Unknown type for TransactionArgument.");
```

Comment:
This isn't true any more, we'll need to start supporting structs soon. We already support strings.

Reply:
By looking at the Move repo, I don't see string as a variant in the type enums.